R/functions-xcmsSwath.R

```diff
@@ -172,8 +172,8 @@
         return(.data2spectra(fromFile = fromFile, rt = x["rt"], polarity = pol,
                              precursorIntensity = x[column],
                              precursorMz = x["mz"], return.type = return.type))
-    cors <- vapply(chr_2@.Data, .correlate_chromatogram, y = chr_1[1, 1],
-                   numeric(1), align = "approx")
+    cors <- vapply(chr_2@.Data, compareChromatograms, y = chr_1[1, 1],
+                   numeric(1), ALIGNFUNARGS = list(method = "approx"))
```

Comment:
Good point. R and BioC don't really agree on that and there is no complete consistency. I also don't really like the ALLCAPS but it's e.g. also used in `lapply` (`X`, `FUN`), `apply` (`MARGIN`). We started using the ALLCAPS thus also in `Spectra` e.g. in `compareSpectra` where we have `FUN` and `MAPFUN` that allow to define functions to be used for similarity calculation respectively mapping of peaks (a la `lapply`).

Now, the problem with `compareSpectra` is that we have `...` to pass additional arguments to **both** functions, the `FUN` and `MAPFUN`. It kinda works, but can be tricky if a parameter with the same name, but different value should be passed to each of the two functions. Thus, the `compareChromatograms` function has parameters `ALIGNFUN` for aligning of the chromatograms and `FUN` to calculate similarity but in addition also the parameters `ALIGNFUNARGS` and `FUNARGS` that allow the user to define arguments specifically for each of the two functions. It's not pretty but efficient...

My (kind of) *consistency* in this inconsistency: 
- I use camelCase argument namings if the argument is used within that function (the default)
- ALLCAPS is used e.g. if a function is expected as the argument to be applied to the data (a la `lapply`)  
- `PARAM` *vs* `ARGS`: for me `PARAM` (or `Param`) implicates a single parameter, and I used it almost exclusively if a parameter **object** is expected (such as `BPPARAM` or `CentWaveParam`). Secondly, I would prefer `ARGS` over `PARMS` because it IMHO reads better - and might be easier to understand for the user.

Reply:
Ok